functional-tests/test-bed/src/test/java/org/arquillian/smart/testing/ftest/newtests/LocalChangesNewTestsSelectionExecutionFunctionalTest.java

```diff
@@ -41,6 +42,31 @@ public void should_only_execute_new_tests_related_to_single_local_change() throw
         assertThat(actualTestResults).containsAll(expectedTestResults).hasSameSizeAs(expectedTestResults);
     }
 
+
+    @Test
+    public void should_only_execute_new_tests_related_to_single_local_change_using_failsafe() {
+
+        // given
+        final Project project = testBed.getProject();
+
+        project.configureSmartTesting()
+            .executionOrder(NEW)
+            .inMode(SELECTING)
+            .useFailsafe()
+            .enable();
+
+        final List<TestResult> expectedTestResults = project
+            .applyAsLocalChanges("Adds new unit test");
+
+        // when
+        final List<TestResult> actualTestResults = project.build("clean", "verify");
```

Comment:
Probably yes, but probably this is not the real case for users who are going to hook failsafe in verify goal.